walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,81 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.sql.Time;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    // private static final int SMALLEST_DURATION = 15;
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        /** the request is too long */
+        if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+            return Arrays.asList();
+        }
```

Comment:
nit: prefers new ArrayList<>();
In this way you simply create an empty ArrayList. Otherwise Arrays.asList() might create an array first then turn into a list. Moreover, it's less obviousl.

Reply:
Done